airflow/contrib/operators/dataproc_operator.py

```diff
@@ -1467,6 +1322,6 @@ def start(self):
             self.hook.get_conn().projects().regions().workflowTemplates()
             .instantiateInline(
                 parent='projects/%s/regions/%s' % (self.project_id, self.region),
-                instanceId=str(uuid.uuid4()),
+                requestId=str(uuid.uuid4()),
```

Comment:
Are you sure it is correct? What was the original problem here and why instanceId was set to random number? I think how it is now is superfluous. RequestId is optional and generating it randomly here makes little sense (it will be generated randomly anyway on the server side). It only makes sense to pass request Id to make sure the operation is idempotent and this will only work when request Id is not randomly generated but passed by the caller explicitly (generated somehow or stored in a local database so that next time we repeat the request we use the same requestID). We need to clarify what we want to achieve here.

Reply:
1 - we're replacing instanceId with requestId. They mean the same thing
2 - you're right that its intended for idempotency. Instead of protecting against Airflow restarts, it guards against first request resulting in HTTP 500 (but succeeding) and then the client retrying and resulting in 2 concurrent dataproc workflows running.

Does Airflow provide any internal id that we could use here instead?

Reply:
1) - I see in the doc instanceId was deprecated and replaced by requestId. so 👍 
2) I also see that the docs recommend using UUID so that's also fine 👍